src/bft.py

```diff
@@ -28,5 +30,13 @@ def bf_order(t: T | None) -> Iterable[int]:
     L = []
     if(t!=None):
         D.append(t)
-        bf_order_D_L(D, L)
+        while D!=deque([]):
+            if(t!=None):
```

Comment:
If you only ever append trees that aren't `None` (you test for it for `left` and `right`, after all), then `t` can never be `None` here.

Reply:
No, because i already tested that 4 lines ago, I do not enter the while loop at all if t is None, I directly return an empty list

Reply:
Well, it could be `None` if you got `None` back from the queue, but since you never put `None` in there, it should never happen.